MiniSpace.Web/src/MiniSpace.Web/Shared/MainLayout.razor

```diff
@@ -5,12 +5,19 @@
 @inject IStudentsService StudentsService
 @inject NavigationManager NavigationManager
 @inject Microsoft.JSInterop.IJSRuntime JSRuntime
+@using Blazored.LocalStorage
+@inject ILocalStorageService localStorage
+@inject AuthenticationStateProvider AuthenticationStateProvider
 
 <RadzenLayout style="width: 100%; min-width: 100vw; min-height: 100vh; text-align: center;" id="radzen-layout">
     <header>
         <RadzenStack Orientation="Orientation.Horizontal" JustifyContent="JustifyContent.SpaceBetween" Style="width: 100%;">
             <RadzenStack Orientation="Orientation.Horizontal" AlignItems="AlignItems.Center" Gap="0">
-                @if (IdentityService.IsAuthenticated && StudentsService.StudentDto.State == "valid")
+                @* 
+                    TODO: do not forget about the code snippet chaned ❕❕❕
+                        @if (IsUserAuthenticated && StudentsService.StudentDto.State == "valid") 
+                *@
+                @if (IsUserAuthenticated) 
                 {
                     <RadzenSidebarToggle Click="@(() => _sidebarExpanded = !_sidebarExpanded)" />
                 }
```

Comment:
Currently I will not resolve this one. Please do not forget about that in the next iteration. This more like 'carving up' with the identity service and the local storage with the students service.